data-model/src/main/java/org/mousephenotype/cda/owl/OntologyParser.java

```diff
@@ -274,18 +314,203 @@ private OntologyTermDTO getDTO(OWLClass cls){
+     * @throws OWLOntologyStorageException
+     */
+    protected Set<String> getTermsInSlim(Set<String> wantedIDs, String prefix)
+            throws IOException, OWLOntologyStorageException {
+
+        // Cache it in termsInSlim so we don't have to re-compute this every time
+        if (termsInSlim != null){
+            return termsInSlim;
+        }
+
+        // Add replacement ids for deprecated classes to wanted ids
+        for (OWLClass cls : ontology.getClassesInSignature()) {
+            if (!cls.getIRI().isNothing() && EntitySearcher.getAnnotations(cls, ontology, TERM_REPLACED_BY) != null && wantedIDs.contains(getIdentifierShortForm(cls))) {
+                for (OWLAnnotation annotation : EntitySearcher.getAnnotations(cls, ontology, TERM_REPLACED_BY)) {
+                    if (annotation.getValue() instanceof OWLLiteral) {
+                        wantedIDs.add(((OWLLiteral) annotation.getValue()).getLiteral());
+                        wantedIDs.remove(getIdentifierShortForm(cls));
+                    }
+                }
+            }
+        }
+
+        // Add the "seed" terms and their ancestors to the slim set.
+        Set<String> classesInSlim = new HashSet<>();
+        for (OWLClass cls : ontology.getClassesInSignature()) {
+            if (wantedIDs.contains(getIdentifierShortForm(cls))) {
+                if (startsWithPrefix(cls, prefix)) {
+                    classesInSlim.add(getIdentifierShortForm(cls));
+                    classesInSlim.addAll(getClassAncestors(cls, null).stream().map(item -> {return getIdentifierShortForm(item);}).collect(Collectors.toSet()));
+                }
+            }
+        }
+
+        termsInSlim = classesInSlim;
+        System.out.println("Set termsInSlim " + termsInSlim.size());
+        return classesInSlim;
+
+    }
+
+
+    /**
+     *
+     * @param cls
+     * @param prefixes
+     * @return
+     */
+    private Set<OWLClass> getClassAncestors(OWLClass cls, Set<String> prefixes){
+
+        if( ancestorsCache.containsKey(getIdentifierShortForm(cls))){
+            return ancestorsCache.get(getIdentifierShortForm(cls));
+        }
+        Set<OWLClass> ancestorIds = new HashSet<>();
+        ancestorIds.addAll(getClassAncestors(cls, prefixes, ancestorIds));
+        ancestorsCache.put(getIdentifierShortForm(cls), ancestorIds);
+
+        return ancestorIds;
+    }
+
 
-        Set<String> eqClasses = new HashSet<>();
-        for (OWLClassExpression classExpression : EntitySearcher.getSubClasses(cls, ontology)){
+    // TODO run reasoner on ontology first
+    /**
+     * This is the recursive method, use the other one as it caches results too
+     * @param cls
+     * @param prefixes
+     * @param ancestorIds
+     * @return
+     */
+    private Set<OWLClass> getClassAncestors(OWLClass cls, Set<String> prefixes, Set<OWLClass> ancestorIds){
+
+        Collection<OWLClassExpression> superClasses = EntitySearcher.getSuperClasses(cls, ontology);
```

Comment:
I thought more about this and I don't think we should use the equivalent classes here because they're not ancestors, they're on the same level.